============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py first driver inits
node_id:3603ea8d
1655733223071,965,selenium_login:open_login_page,Success,True
1655733224522,1369,selenium_login:login_and_view_dashboard,Success,True
1655733224523,2416,selenium_login,Success,True
.test_0_selenium_a_login,Total:1 Passed:1 Failed:0
1655733225508,950,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:2 Passed:2 Failed:0
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34994098', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9766720', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34994098', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9766720', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:3 Passed:2 Failed:1
1655733226415,363,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:4 Passed:3 Failed:1


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34994098', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9766720', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34994098', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9766720', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 5.17s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733226690,186,selenium_login:open_login_page,Success,True
1655733227092,330,selenium_login:login_and_view_dashboard,Success,True
1655733227092,589,selenium_login,Success,True
.test_0_selenium_a_login,Total:5 Passed:4 Failed:1
1655733227855,717,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:6 Passed:5 Failed:1
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25852510', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9389425', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25852510', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9389425', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:7 Passed:5 Failed:2
1655733228587,156,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:8 Passed:6 Failed:2


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25852510', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9389425', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25852510', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9389425', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 2.11s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733228809,135,selenium_login:open_login_page,Success,True
1655733229218,338,selenium_login:login_and_view_dashboard,Success,True
1655733229218,544,selenium_login,Success,True
.test_0_selenium_a_login,Total:9 Passed:7 Failed:2
1655733229619,376,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:10 Passed:8 Failed:2
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25829140', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9278494', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25829140', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9278494', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:11 Passed:8 Failed:3
1655733230210,151,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:12 Passed:9 Failed:3


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25829140', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9278494', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25829140', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9278494', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.56s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733230426,130,selenium_login:open_login_page,Success,True
1655733230811,316,selenium_login:login_and_view_dashboard,Success,True
1655733230811,515,selenium_login,Success,True
.test_0_selenium_a_login,Total:13 Passed:10 Failed:3
1655733231421,585,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:14 Passed:11 Failed:3
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34974700', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9785621', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34974700', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9785621', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:15 Passed:11 Failed:4
1655733232032,155,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:16 Passed:12 Failed:4


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34974700', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9785621', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34974700', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9785621', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.76s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733232251,135,selenium_login:open_login_page,Success,True
1655733232646,324,selenium_login:login_and_view_dashboard,Success,True
1655733232646,530,selenium_login,Success,True
.test_0_selenium_a_login,Total:17 Passed:13 Failed:4
1655733233840,1140,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:18 Passed:14 Failed:4
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '26116085', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9777784', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '26116085', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9777784', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:19 Passed:14 Failed:5
1655733234487,197,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:20 Passed:15 Failed:5


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '26116085', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9777784', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '26116085', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9777784', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 2.40s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733234746,156,selenium_login:open_login_page,Success,True
1655733235361,544,selenium_login:login_and_view_dashboard,Success,True
1655733235361,771,selenium_login,Success,True
.test_0_selenium_a_login,Total:21 Passed:16 Failed:5
1655733236171,778,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:22 Passed:17 Failed:5
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34971257', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9782237', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34971257', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9782237', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:23 Passed:17 Failed:6
1655733236979,242,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:24 Passed:18 Failed:6


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34971257', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9782237', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34971257', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9782237', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 2.41s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733237204,137,selenium_login:open_login_page,Success,True
1655733237674,401,selenium_login:login_and_view_dashboard,Success,True
1655733237675,608,selenium_login,Success,True
.test_0_selenium_a_login,Total:25 Passed:19 Failed:6
1655733238344,619,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:26 Passed:20 Failed:6
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34975440', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9788512', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34975440', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9788512', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:27 Passed:20 Failed:7
1655733239087,231,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:28 Passed:21 Failed:7


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34975440', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9788512', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34975440', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9788512', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 2.05s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733239312,142,selenium_login:open_login_page,Success,True
1655733239717,336,selenium_login:login_and_view_dashboard,Success,True
1655733239717,546,selenium_login,Success,True
.test_0_selenium_a_login,Total:29 Passed:22 Failed:7
1655733240384,599,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:30 Passed:23 Failed:7
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '35147270', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9164709', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '35147270', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9164709', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:31 Passed:23 Failed:8
1655733241246,252,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:32 Passed:24 Failed:8


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '35147270', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9164709', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '35147270', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9164709', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 2.10s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733241532,201,selenium_login:open_login_page,Success,True
1655733242009,407,selenium_login:login_and_view_dashboard,Success,True
1655733242009,678,selenium_login,Success,True
.test_0_selenium_a_login,Total:33 Passed:25 Failed:8
1655733242979,938,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:34 Passed:26 Failed:8
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34993597', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9172047', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34993597', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9172047', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:35 Passed:26 Failed:9
1655733243672,159,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:36 Passed:27 Failed:9


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34993597', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9172047', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34993597', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9172047', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 2.37s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733243892,133,selenium_login:open_login_page,Success,True
1655733244296,334,selenium_login:login_and_view_dashboard,Success,True
1655733244296,537,selenium_login,Success,True
.test_0_selenium_a_login,Total:37 Passed:28 Failed:9
1655733244733,369,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:38 Passed:29 Failed:9
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34990837', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9173599', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34990837', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9173599', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:39 Passed:29 Failed:10
1655733245296,157,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:40 Passed:30 Failed:10


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34990837', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9173599', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34990837', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9173599', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.57s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733245504,124,selenium_login:open_login_page,Success,True
1655733246006,390,selenium_login:login_and_view_dashboard,Success,True
1655733246006,627,selenium_login,Success,True
.test_0_selenium_a_login,Total:41 Passed:31 Failed:10
1655733246766,728,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:42 Passed:32 Failed:10
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '35151345', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9301807', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '35151345', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9301807', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:43 Passed:32 Failed:11
1655733247316,150,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:44 Passed:33 Failed:11


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '35151345', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9301807', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '35151345', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9301807', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.96s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733247571,172,selenium_login:open_login_page,Success,True
1655733247946,307,selenium_login:login_and_view_dashboard,Success,True
1655733247946,547,selenium_login,Success,True
.test_0_selenium_a_login,Total:45 Passed:34 Failed:11
1655733249054,1074,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:46 Passed:35 Failed:11
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '35141395', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9284372', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '35141395', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9284372', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:47 Passed:35 Failed:12
1655733249573,143,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:48 Passed:36 Failed:12


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '35141395', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9284372', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '35141395', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9284372', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 2.20s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733249851,168,selenium_login:open_login_page,Success,True
1655733250250,330,selenium_login:login_and_view_dashboard,Success,True
1655733250251,568,selenium_login,Success,True
.test_0_selenium_a_login,Total:49 Passed:37 Failed:12
1655733250906,623,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:50 Passed:38 Failed:12
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34993812', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9291408', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34993812', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9291408', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:51 Passed:38 Failed:13
1655733251559,188,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:52 Passed:39 Failed:13


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34993812', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9291408', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34993812', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9291408', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.90s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733251770,127,selenium_login:open_login_page,Success,True
1655733252177,337,selenium_login:login_and_view_dashboard,Success,True
1655733252177,534,selenium_login,Success,True
.test_0_selenium_a_login,Total:53 Passed:40 Failed:13
1655733252763,545,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:54 Passed:41 Failed:13
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34975201', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9285095', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34975201', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9285095', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:55 Passed:41 Failed:14
1655733253472,169,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:56 Passed:42 Failed:14


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34975201', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9285095', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34975201', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9285095', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.86s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733253681,125,selenium_login:open_login_page,Success,True
1655733254065,315,selenium_login:login_and_view_dashboard,Success,True
1655733254065,509,selenium_login,Success,True
.test_0_selenium_a_login,Total:57 Passed:43 Failed:14
1655733254670,571,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:58 Passed:44 Failed:14
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25984655', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9297995', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25984655', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9297995', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:59 Passed:44 Failed:15
1655733255242,142,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:60 Passed:45 Failed:15


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25984655', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9297995', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25984655', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9297995', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.71s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733255452,125,selenium_login:open_login_page,Success,True
1655733255836,315,selenium_login:login_and_view_dashboard,Success,True
1655733255836,509,selenium_login,Success,True
.test_0_selenium_a_login,Total:61 Passed:46 Failed:15
1655733256477,606,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:62 Passed:47 Failed:15
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '35155287', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9291477', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '35155287', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9291477', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:63 Passed:47 Failed:16
1655733257086,137,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:64 Passed:48 Failed:16


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '35155287', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9291477', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '35155287', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9291477', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.78s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733257345,176,selenium_login:open_login_page,Success,True
1655733257726,312,selenium_login:login_and_view_dashboard,Success,True
1655733257727,558,selenium_login,Success,True
.test_0_selenium_a_login,Total:65 Passed:49 Failed:16
1655733258317,553,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:66 Passed:50 Failed:16
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25830904', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9382408', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25830904', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9382408', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:67 Passed:50 Failed:17
1655733258869,152,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:68 Passed:51 Failed:17


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25830904', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9382408', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25830904', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9382408', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.73s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733259122,168,selenium_login:open_login_page,Success,True
1655733259489,299,selenium_login:login_and_view_dashboard,Success,True
1655733259490,536,selenium_login,Success,True
.test_0_selenium_a_login,Total:69 Passed:52 Failed:17
1655733260127,601,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:70 Passed:53 Failed:17
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25956650', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9781603', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25956650', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9781603', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:71 Passed:53 Failed:18
1655733260681,148,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:72 Passed:54 Failed:18


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25956650', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9781603', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25956650', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9781603', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.75s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733260951,184,selenium_login:open_login_page,Success,True
1655733261310,290,selenium_login:login_and_view_dashboard,Success,True
1655733261310,543,selenium_login,Success,True
.test_0_selenium_a_login,Total:73 Passed:55 Failed:18
1655733261877,530,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:74 Passed:56 Failed:18
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25967082', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9281908', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25967082', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9281908', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:75 Passed:56 Failed:19
1655733262495,128,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:76 Passed:57 Failed:19


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25967082', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9281908', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25967082', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9281908', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.76s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733262747,168,selenium_login:open_login_page,Success,True
1655733263110,292,selenium_login:login_and_view_dashboard,Success,True
1655733263110,531,selenium_login,Success,True
.test_0_selenium_a_login,Total:77 Passed:58 Failed:19
1655733263711,562,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:78 Passed:59 Failed:19
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34983266', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9274316', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34983266', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9274316', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:79 Passed:59 Failed:20
1655733264311,147,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:80 Passed:60 Failed:20


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34983266', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9274316', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34983266', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9274316', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.76s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733264556,133,selenium_login:open_login_page,Success,True
1655733264987,325,selenium_login:login_and_view_dashboard,Success,True
1655733264987,564,selenium_login,Success,True
.test_0_selenium_a_login,Total:81 Passed:61 Failed:20
1655733265523,496,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:82 Passed:62 Failed:20
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25823493', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9391814', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25823493', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9391814', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:83 Passed:62 Failed:21
1655733266083,129,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:84 Passed:63 Failed:21


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25823493', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9391814', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25823493', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9391814', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.69s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733266289,122,selenium_login:open_login_page,Success,True
1655733266697,304,selenium_login:login_and_view_dashboard,Success,True
1655733266697,531,selenium_login,Success,True
.test_0_selenium_a_login,Total:85 Passed:64 Failed:21
1655733267257,520,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:86 Passed:65 Failed:21
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34976340', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9376590', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34976340', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9376590', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:87 Passed:65 Failed:22
1655733267803,130,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:88 Passed:66 Failed:22


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34976340', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9376590', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34976340', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9376590', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.66s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733268050,164,selenium_login:open_login_page,Success,True
1655733268414,295,selenium_login:login_and_view_dashboard,Success,True
1655733268414,528,selenium_login,Success,True
.test_0_selenium_a_login,Total:89 Passed:67 Failed:22
1655733269016,561,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:90 Passed:68 Failed:22
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25823148', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9284313', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25823148', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9284313', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:91 Passed:68 Failed:23
1655733269568,135,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:92 Passed:69 Failed:23


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25823148', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9284313', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25823148', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9284313', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.71s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733269821,170,selenium_login:open_login_page,Success,True
1655733270201,312,selenium_login:login_and_view_dashboard,Success,True
1655733270201,551,selenium_login,Success,True
.test_0_selenium_a_login,Total:93 Passed:70 Failed:23
1655733270815,574,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:94 Passed:71 Failed:23
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25955249', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9169257', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25955249', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9169257', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:95 Passed:71 Failed:24
1655733271418,149,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:96 Passed:72 Failed:24


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25955249', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9169257', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25955249', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9169257', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.79s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733271662,166,selenium_login:open_login_page,Success,True
1655733272024,291,selenium_login:login_and_view_dashboard,Success,True
1655733272025,529,selenium_login,Success,True
.test_0_selenium_a_login,Total:97 Passed:73 Failed:24
1655733272586,521,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:98 Passed:74 Failed:24
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25834968', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '11512981', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25834968', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '11512981', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:99 Passed:74 Failed:25
1655733273180,147,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:100 Passed:75 Failed:25


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25834968', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '11512981', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25834968', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '11512981', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.70s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733273430,167,selenium_login:open_login_page,Success,True
1655733273789,290,selenium_login:login_and_view_dashboard,Success,True
1655733273789,526,selenium_login,Success,True
.test_0_selenium_a_login,Total:101 Passed:76 Failed:25
1655733274480,650,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:102 Passed:77 Failed:25
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25954751', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9778531', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25954751', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9778531', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:103 Passed:77 Failed:26
1655733275123,191,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:104 Passed:78 Failed:26


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25954751', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9778531', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25954751', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9778531', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.89s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733275332,125,selenium_login:open_login_page,Success,True
1655733275690,288,selenium_login:login_and_view_dashboard,Success,True
1655733275690,484,selenium_login,Success,True
.test_0_selenium_a_login,Total:105 Passed:79 Failed:26
1655733276260,528,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:106 Passed:80 Failed:26
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25965258', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9795145', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25965258', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9795145', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:107 Passed:80 Failed:27
1655733276910,189,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:108 Passed:81 Failed:27


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25965258', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9795145', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25965258', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9795145', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.73s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733277124,128,selenium_login:open_login_page,Success,True
1655733277501,310,selenium_login:login_and_view_dashboard,Success,True
1655733277501,506,selenium_login,Success,True
.test_0_selenium_a_login,Total:109 Passed:82 Failed:27
1655733278097,551,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:110 Passed:83 Failed:27
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25841680', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9381422', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25841680', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9381422', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:111 Passed:83 Failed:28
1655733278664,142,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:112 Passed:84 Failed:28


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '25841680', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9381422', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '25841680', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9381422', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.70s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733278876,128,selenium_login:open_login_page,Success,True
1655733279316,368,selenium_login:login_and_view_dashboard,Success,True
1655733279317,569,selenium_login,Success,True
.test_0_selenium_a_login,Total:113 Passed:85 Failed:28
1655733279894,525,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:114 Passed:86 Failed:28
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34984130', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9385216', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34984130', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9385216', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:115 Passed:86 Failed:29
1655733280609,196,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:116 Passed:87 Failed:29


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '34984130', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9385216', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '34984130', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9385216', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.89s ====================
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /dc-app-performance-toolkit/app
plugins: forked-1.4.0, xdist-2.5.0, apiritif-1.1.2
collected 4 items

selenium_ui/confluence_ui.py get driver from global
clear browser cookies
node_id:3603ea8d
1655733280827,130,selenium_login:open_login_page,Success,True
1655733281218,318,selenium_login:login_and_view_dashboard,Success,True
1655733281218,521,selenium_login,Success,True
.test_0_selenium_a_login,Total:117 Passed:88 Failed:29
1655733281819,553,selenium_view_page,Success,True
.test_1_selenium_view_page,Total:118 Passed:89 Failed:29
FAction: test_1_selenium_custom_action, Error: confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '35228155', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9794840', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '35228155', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9794840', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError

test_1_selenium_custom_action,Total:119 Passed:89 Failed:30
1655733282347,139,selenium_log_out,Success,True
.test_2_selenium_z_log_out,Total:120 Passed:90 Failed:30


=================================== FAILURES ===================================
________________________ test_1_selenium_custom_action _________________________

confluence_webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
confluence_datasets = {'blog_id': '35228155', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9794840', ...}
confluence_screen_shots = None

    def test_1_selenium_custom_action(confluence_webdriver, confluence_datasets, confluence_screen_shots):
>        extension_ui.app_specific_action(confluence_webdriver, confluence_datasets)

selenium_ui/confluence_ui.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

webdriver = <selenium.webdriver.chrome.webdriver.WebDriver (session="07e0ef937f25429abae3d1b17091473f")>
datasets = {'blog_id': '35228155', 'blogs': [['25982229', 'WCS'], ['25720224', 'NFVB'], ['25847101', 'PRCJA'], ['25954597', 'DPPN'], ['25963247', 'AMN'], ['25829541', 'ODBAR'], ...], 'custom_pages': [], 'page_id': '9794840', ...}

    def app_specific_action(webdriver, datasets):
        page = BasePage(webdriver)
        if datasets['custom_pages']:
            app_specific_page_id = datasets['custom_page_id']
    
        mykeys=','.join(str(e) for e in datasets['custom_pages'])
>       print(f"Custom page  {datasets['custom_page_id']} form spacekey {mykeys}")
E       KeyError: 'custom_page_id'

extension/confluence/extension_ui.py:17: KeyError
=============================== warnings summary ===============================
../../root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114
  /root/.bzt/python-packages/3.9.5/_pytest/config/__init__.py:1114: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: apiritif
    self._mark_plugins_for_rewrite(hook)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED selenium_ui/confluence_ui.py::test_1_selenium_custom_action - KeyError...
==================== 1 failed, 3 passed, 1 warning in 1.68s ====================
